lib/rubysl/rubysl-strscan/lib/rubysl/strscan/strscan.rb

```diff
@@ -125,6 +125,8 @@ def initialize(string, dup=false)
       @string = String.new @original
     end
 
+    @prev_pos = @pos = @width = 0
+    @match = nil
```

Comment:
The `reset_state` call just below does the same thing as these 2 lines, what is the advantage of duplicating them?